src/components/bootcamp/EditInternForm.tsx

```diff
@@ -0,0 +1,143 @@
+import React from 'react'
+import * as yup from 'yup'
+import { useRouter } from 'next/router'
+import { FormikHelpers } from 'formik'
+import { useMutation } from 'react-query'
+import { Button, Grid, Typography } from '@mui/material'
+import { AxiosError, AxiosResponse } from 'axios'
+import { useTranslation } from 'next-i18next'
+import { makeStyles } from '@mui/styles'
+
+import { routes } from 'common/routes'
+import { AlertStore } from 'stores/AlertStore'
+import { editBootcampIntern } from 'common/rest'
+import { BootcampFormData, BootcampInput, BootcampResponse } from 'gql/bootcamp'
+import GenericForm from 'components/common/form/GenericForm'
+import FormTextField from 'components/common/form/FormTextField'
+import { name, phone } from 'common/form/validation'
+import { ApiErrors, isAxiosError, matchValidator } from 'common/api-errors'
+import LinkButton from 'components/common/LinkButton'
+
+const useStyles = makeStyles({
+  container: {
+    maxWidth: '700px',
+    margin: '0 auto',
+  },
+})
+
+const validationSchema: yup.SchemaOf<BootcampFormData> = yup.object().defined().shape({
+  firstName: name.required(),
+  lastName: name.required(),
```

Comment:
You can use the standard `name`  validator from `common/validators` to achieve the same structure and reusability